install/install.sh

```diff
@@ -45,328 +45,362 @@ error() {
   printf "%s\n" "${RED}x $*${NO_COLOR}" >&2
 }
 
-complete() {
+success() {
   printf "%s\n" "${GREEN}✓${NO_COLOR} $*"
 }
 
-# Gets path to a temporary file, even if
-get_tmpfile() {
-  local suffix
-  suffix="$1"
-  if hash mktemp; then
-    printf "%s.%s" "$(mktemp)" "${suffix}"
-  else
-    # No really good options here--let's pick a default + hope
-    printf "/tmp/starship.%s" "${suffix}"
-  fi
+has() {
+  command -v "$1" >/dev/null 2>&1
 }
 
-# Test if a location is writeable by trying to write to it. Windows does not let
-# you test writeability other than by writing: https://stackoverflow.com/q/1999988
-test_writeable() {
-  local path
-  path="${1:-}/test.txt"
-  if touch "${path}" 2>/dev/null; then
-    rm "${path}"
-    return 0
-  else
+mktmpdir() {
+  tmpdir=$(mktemp -dqt "$1") || {
+    error "Can't create temp file, exiting…"
     return 1
-  fi
+  }
+  echo "$tmpdir"
 }
 
 fetch() {
-  local command
-  if hash curl 2>/dev/null; then
-    set +e
-    command="curl --silent --fail --location $1"
-    curl --silent --fail --location "$1"
-    rc=$?
-    set -e
-  else
-    if hash wget 2>/dev/null; then
-      set +e
-      command="wget -O- -q $1"
-      wget -O- -q "$1"
-      rc=$?
-      set -e
+  file="$1"
+  url="$2"
+
+  if has curl; then
+    cmd="curl -w '%{http_code}' -sL -o $file $url"
```

Comment:
I've c/p it 🤣

Reply:
Let's at least convert this to long-flag form then. It won't be perfect, but I think `curl --write-out '${http_code}' --silent --location --output $file $url` at least gives a few more hints as to what each flag is doing, and is slightly easier to look up in the manpage if needed.

Reply:
Printing out HTTP code is actually redundant since we are going back to the `--fail` approach. Regarding other flags, well, I've converted them to short ones in order to make diagnostic output (error trace) shorter. TBH I don't care much about their form, so I'll just change it as you suggested.